netket/models/autoreg.py

```diff
@@ -112,6 +112,15 @@ def conditional(self, inputs: Array, index: int) -> Array:
         Returns:
           The probabilities with dimensions (batch, Hilbert.local_size).
         """
+        # TODO: remove this in future
+        if hasattr(self, "_conditional"):
+            from netket.utils import warn_deprecation
+
+            warn_deprecation(
+                "AbstractARNN._conditional has been renamed to AbstractARNN.conditional "
+                "as a public API. Please update you code to use fast AR sampling."
+            )
+
```

Comment:
I think you should return `self._conditional(inputs, index)` here if it's a deprecation, so that it still supports the old syntax

Reply:
Ok I agree. We also renamed `_reshape_inputs` to `reshape_inputs`, but previously it was a very internal undocumented helper function, and without it the code will fail loudly, so I think we don't need to carefully deprecate it.

In the case of `conditional`, it will silently choose the slow sampling if we don't add the deprecation.

Reply:
putting the deprecation in conditional (with a `#TODO: deprecated in 3.7, November 2022` ) is ok. I also have some code that defined `_conditional` so it would be helpful if it kept working. Reshape input was super internal so I don't mind the change.